pandas/core/arrays/numpy_.py

```diff
@@ -278,6 +278,9 @@ def fillna(self, value=None, method=None, limit=None):
         return new_values
 
     def take(self, indices, allow_fill=False, fill_value=None):
+        if fill_value is None:
+            # Primarily for subclasses
+            fill_value = self.dtype.na_value
```

Comment:
why is this necessary?

Reply:
For StringArray.take. It wasn't needed before since PandasArray uses `np.nan` as its na_value, which is the default used in take algos.